ethereum/solidity/contracts/IDAOracle.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./MessageTuple.sol";
+import "./lib/tree/binary/BinaryMerkleProof.sol";
+
+/// @notice Data Availability Oracle interface.
+interface IDAOracle {
+    /// @notice Verify a Data Availability attestation.
+    /// @param _tupleRootIndex Index of the tuple root to prove against.
+    /// @param _tuple Message tuple to prove inclusion of.
+    /// @param _proof Binary Merkle tree proof that `tuple` is in the root at `tupleRootIndex`.
+    /// @return `true` is proof is valid, `false` otherwise.
+    function verifyAttestation(
+        uint256 _tupleRootIndex,
+        MessageTuple memory _tuple,
```

Comment:
This is obsolete now with #42 ? Can this PR be closed now? Also what is with #32? That one should be an NMT now, right?

Reply:
I'm going to modify this PR to use a DA root instead of message tuple root. Don't want to discard the rest of the comments and interface.